src/input_xml.F90

```diff
@@ -2578,20 +2298,38 @@ subroutine read_multipole_data(i_table)
     logical :: file_exists                 ! Does multipole library exist?
     character(7) :: readable               ! Is multipole library readable?
     character(MAX_FILE_LEN) :: filename    ! Path to multipole xs library
+    character(kind=C_CHAR), pointer :: string(:)
     integer(HID_T) :: file_id
     integer(HID_T) :: group_id
 
-    ! For the time being, and I know this is a bit hacky, we just assume
-    ! that the file will be ZZZAAAmM.h5.
+    interface
+      function path_multipole_c() result(ptr) bind(C)
+        import C_PTR
+        type(C_PTR) :: ptr
+      end function
+    end interface
+
     associate (nuc => nuclides(i_table))
 
-      if (nuc % metastable > 0) then
-        filename = trim(path_multipole) // trim(zero_padded(nuc % Z, 3)) // &
-             trim(zero_padded(nuc % A, 3)) // 'm' // &
-             trim(to_str(nuc % metastable)) // ".h5"
+      if (library_present(LIBRARY_WMP, to_lower(nuc % name))) then
+        ! If WMP data is listed in cross_sections.xml, prefer that
+        filename = library_path(LIBRARY_WMP, to_lower(nuc % name))
       else
-        filename = trim(path_multipole) // trim(zero_padded(nuc % Z, 3)) // &
-             trim(zero_padded(nuc % A, 3)) // ".h5"
+        ! Otherwise, we rely on the OPENMC_MULTIPOLE_LIBRARY environment
+        ! variable. This is a bit hacky, but we just assume that the file will be
+        ! ZZZAAAmM.h5.
+
+        call c_f_pointer(path_multipole_c(), string, [255])
+        path_multipole = to_f_string(string)
+
+        if (nuc % metastable > 0) then
+          filename = trim(path_multipole) // trim(zero_padded(nuc % Z, 3)) // &
+               trim(zero_padded(nuc % A, 3)) // 'm' // &
+               trim(to_str(nuc % metastable)) // ".h5"
+        else
+          filename = trim(path_multipole) // trim(zero_padded(nuc % Z, 3)) // &
+               trim(zero_padded(nuc % A, 3)) // ".h5"
+        end if
```

Comment:
Are you planning to remove the 'hacky' way of looking for the wmp library?

Reply:
The new way is definitely clearer and simpler. And I found the new way makes it easier to check if a wmp library exists before reading the point-wise library, this will be helpful for implementation of elastic scattering using multipole data instead of 0K point-wise data. Therefore, I prefer removing the old way. The only downside of to me is the discarding of `OPENMC_MULTIPOLE_LIBRARY`, the users may need to update the `cross_sections.xml` manually to include the wmp library. The travis CI should do this step too.